app/src/main/java/de/westnordost/streetcomplete/quests/recycling_material/AddRecyclingContainerMaterials.kt

```diff
@@ -0,0 +1,52 @@
+package de.westnordost.streetcomplete.quests.recycling_material
+
+import de.westnordost.osmapi.map.data.BoundingBox
+import de.westnordost.osmapi.map.data.Element
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao
+import de.westnordost.streetcomplete.data.osm.tql.getQuestPrintStatement
+import de.westnordost.streetcomplete.data.osm.tql.toGlobalOverpassBBox
+
+class AddRecyclingContainerMaterials(private val overpassServer: OverpassMapDataDao) : OsmElementQuestType<List<String>> {
+    override val commitMessage = "Add recycled materials to container"
+    override val icon = R.drawable.ic_quest_recycling_materials
+
+    override fun download(bbox: BoundingBox, handler: MapDataWithGeometryHandler): Boolean {
+        return overpassServer.getAndHandleQuota(getOverpassQuery(bbox), handler)
+    }
+
+    private fun getOverpassQuery(bbox: BoundingBox) = """
+        ${bbox.toGlobalOverpassBBox()}
+        node[amenity = recycling][recycling_type = container][~"^recycling:.*$"~".*"] -> .known;
+        node[amenity = recycling][recycling_type = container] -> .all;
```

Comment:
I don't know how much it would save, but is it possible to swap these, filtering the .all set for the .known set? Something like ```node.all[~"^recycling:.*$"~".*"] -> .known;``` ?

Reply:
Yes, thanks!